File: source/2.两数相加.cpp
/*
 * @lc app=leetcode.cn id=2 lang=cpp
 *
 * [2] 两数相加
 */
#include "cpp_includes.h"
// @lc code=start
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
 public:
  ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
    ListNode dummyNode;
    ListNode* cur = &dummyNode;

    int sum = 0;
    while (l1 || l2 || sum) {
      if (l1) sum += l1->val;
      if (l2) sum += l2->val;
      cur->next = new ListNode(sum % 10);
      sum /= 10;
      cur = cur->next;
      if (l1) l1 = l1->next;
      if (l2) l2 = l2->next;
    }
    return dummyNode.next;
  }
};
// @lc code=end

File: source/2.两数相加.cs
/*
 * @lc app=leetcode.cn id=2 lang=csharp
 *
 * [2] 两数相加
 */

// @lc code=start
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution
{
  public ListNode AddTwoNumbers(ListNode l1, ListNode l2)
  {
    var dummy = new ListNode();
    var cur = dummy;

    int sum = 0;
    while (l1 != null || l2 != null || sum != 0)
    {
      if (l1 != null) sum += l1.val;
      if (l2 != null) sum += l2.val;

      cur.next = new ListNode(sum % 10);
      sum /= 10;
      cur = cur.next;

      if (l1 != null) l1 = l1.next;
      if (l2 != null) l2 = l2.next;
    }
    return dummy.next;
  }
}
// @lc code=end

